sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -71,7 +71,9 @@ class DataFrameReader private[sql](sqlContext: SQLContext) extends Logging {
    * @since 1.4.0
    */
   def option(key: String, value: String): DataFrameReader = {
-    this.extraOptions += (key -> value)
+    if (value != null) {
```

Comment:
Passing null into the option can cause null pointer exception in data source like spark-csv. I am not sure which option you mean. But `CSVOption` doesn't handle null in particular as I see. It just handles the case without a value (i.e. default value is used then).